src/charm.py

```diff
@@ -178,9 +178,13 @@ def _configure(self, _):
         # Make sure that if the remote_write endpoint changes, it is reflected in relation data.
         self.remote_write_provider.update_endpoint()
         self.grafana_source_provider.update_source(self._external_url)
-
+        self.unit.set_workload_version(self._prometheus_server.version())
         self.unit.status = ActiveStatus()
 
+    def _update_status(self, event):
+        """Fired intermittently by the Juju agent."""
+        self.unit.set_workload_version(self._prometheus_server.version())
```

Comment:
Is `update-status` added in case the workload app is ready on `pebble-ready`?

Reply:
Yeh, it's essentially a catch all in case for whatever reason Prometheus isn't quite ready at the end of the pebble-ready hook. The implementation of `version` on the `Prometheus` server provides no guarantees